src/Helper/FasitHelper.php

```diff
@@ -0,0 +1,478 @@
+    $certificateOptions =
+      $certificateLocator->hasPassphrase() ?
+        [$localCertFilename, $certificateLocator->getPassphrase()]
+        : $localCertFilename;
+
+    return [$certificateOptions, $localCertFilename];
+  }
+
+  /**
+   * Uploads attachment to Fasit.
+   *
+   * @param string $submissionId
+   *   The submission id.
+   * @param array $handlerConfiguration
+   *   The Fasit handler configuration.
+   *
+   * @throws \Drupal\Component\Plugin\Exception\InvalidPluginDefinitionException
+   * @throws \Drupal\Component\Plugin\Exception\PluginNotFoundException
+   * @throws \Drupal\os2forms_fasit\Exception\CertificateLocatorException
+   * @throws \Drupal\os2forms_fasit\Exception\FasitResponseException
+   * @throws \Drupal\os2forms_fasit\Exception\InvalidSettingException
+   *
+   * @phpstan-param array<string, mixed> $handlerConfiguration
+   * @phpstan-return array<string, mixed>
+   */
+  private function uploadAttachment(string $submissionId, array $handlerConfiguration): array {
+    /** @var \Drupal\webform\Entity\WebformSubmission $submission */
+    $submission = $this->getSubmission($submissionId);
+
+    $this->checkHandlerConfiguration($handlerConfiguration, FasitWebformHandler::FASIT_HANDLER_ATTACHMENT_ELEMENT);
+
+    $webformAttachmentElementId = $handlerConfiguration[FasitWebformHandler::FASIT_HANDLER_GENERAL][FasitWebformHandler::FASIT_HANDLER_ATTACHMENT_ELEMENT];
+    $webformAttachmentElement = $submission->getWebform()->getElement($webformAttachmentElementId);
+
+    if ('pdf' != $webformAttachmentElement['#export_type']) {
+      throw new InvalidSettingException(sprintf('Export type of attachment element (%s) must be pdf, found %s', $webformAttachmentElementId, $webformAttachmentElement['#export_type']));
+    }
+
+    $fileContent = AttachmentElement::getFileContent($webformAttachmentElement, $submission);
+    $fileName = AttachmentElement::getFileName($webformAttachmentElement, $submission);
+    $tempAttachmentFilename = tempnam(sys_get_temp_dir(), 'attachment');
+    file_put_contents($tempAttachmentFilename, $fileContent);
+
+    return $this->uploadFile($fileName, $tempAttachmentFilename);
+  }
+
+  /**
+   * Uploads file to Fasit.
+   *
+   * @param string $originalFilename
+   *   The original filename.
+   * @param string $tempFilename
+   *   The temp filename.
+   *
+   * @throws \Drupal\os2forms_fasit\Exception\CertificateLocatorException
+   *   Certificate locator exception.
+   * @throws \Drupal\os2forms_fasit\Exception\FasitResponseException
+   *   Fasit response exception.
+   *
+   * @phpstan-return array<string, mixed>
+   */
+  private function uploadFile(string $originalFilename, string $tempFilename): array {
+    // Endpoint: {baseurl}/{tenant}/{version}/documents/{method}.
+    $endpoint = $this->settings->getFasitApiBaseUrl() . '/' . $this->settings->getFasitApiTenant() . '/' . $this->settings->getFasitApiVersion() . '/documents/' . self::FASIT_API_METHOD_UPLOAD;
```

Comment:
Consider cleaning up the code to make it more clean what you're doing rather than adding a comment to tell it, e.g.

```suggestion
    $endpoint = sprintf('%s/%s/%s/documents/%s', 
      $this->settings->getFasitApiBaseUrl(),
      $this->settings->getFasitApiTenant(),
      $this->settings->getFasitApiVersion(),
      self::FASIT_API_METHOD_UPLOAD
    );
```

Reply:
👍